### PR TITLE
fix: forget to call continuation.finish() causing task suspension

### PR DESCRIPTION
**Pull request by random-yang**
```swift
import EventSource

let eventSource = EventSource()
let dataTask = eventSource.dataTask(for: urlRequest)

for await event in dataTask.events() {
}

// Code blow will never be fired.
// And it will lead to memory leaks.
print("finish") // will not fire
```

### CODE REVIEW

**Comment by Recouse** [comment]
Hey! Thanks for the PR. I'll check it out and merge ASAP

File: Sources/EventSource/EventSource.swift
Comment (Recouse) [inline Sources/EventSource/EventSource.swift:217]: Finishing the continuation here is redundant, since `close()` is always called after `sendErrorEvent(with:)`

**Review by Recouse — CHANGES_REQUESTED** [review]

File: Sources/EventSource/EventSource.swift
Comment (random-yang) [reply Sources/EventSource/EventSource.swift]: You are right. I will rm it.

**Comment by Recouse** [comment]
Thank you for the change. During testing, I caught a bug: `EventType.closed` is never sent on errors. I’ll merge this for now and will fix the bug after the merge

**Review by Recouse — APPROVED** [review]

**Pull request merged**
